google/cloud/spanner_dbapi/cursor.py

```diff
@@ -366,48 +368,41 @@ def __init__(
         self.scale = scale
         self.null_ok = null_ok
 
+        self.fields = (
+            self.name,
+            self.type_code,
+            self.display_size,
+            self.internal_size,
+            self.precision,
+            self.scale,
+            self.null_ok,
+        )
+
     def __repr__(self):
         return self.__str__()
 
     def __getitem__(self, index):
-        if index == 0:
-            return self.name
-        elif index == 1:
-            return self.type_code
-        elif index == 2:
-            return self.display_size
-        elif index == 3:
-            return self.internal_size
-        elif index == 4:
-            return self.precision
-        elif index == 5:
-            return self.scale
-        elif index == 6:
-            return self.null_ok
+        return self.fields[index]
 
     def __str__(self):
-        rstr = ", ".join(
-            [
-                field
-                for field in [
+        str_repr = ", ".join(
+            filter(
+                lambda part: part is not None,
+                [
                     "name='%s'" % self.name,
                     "type_code=%d" % self.type_code,
-                    None
-                    if not self.display_size
-                    else "display_size=%d" % self.display_size,
-                    None
-                    if not self.internal_size
-                    else "internal_size=%d" % self.internal_size,
-                    None
-                    if not self.precision
-                    else "precision='%s'" % self.precision,
-                    None if not self.scale else "scale='%s'" % self.scale,
-                    None
-                    if not self.null_ok
-                    else "null_ok='%s'" % self.null_ok,
-                ]
-                if field
-            ]
+                    "display_size=%d" % self.display_size
+                    if self.display_size
+                    else None,
+                    "internal_size=%d" % self.internal_size
+                    if self.internal_size
+                    else None,
+                    "precision='%s'" % self.precision
+                    if self.precision
+                    else None,
+                    "scale='%s'" % self.scale if self.scale else None,
+                    "null_ok='%s'" % self.null_ok if self.null_ok else None,
+                ],
+            )
```

Comment:
I think with `filter` and inversing `not` statements, it'll be little bit easier to read this code